<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>4 - Running a simple simulation with a pathogen with mixed reproduction system</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">4 - Running a simple simulation with a
pathogen with mixed reproduction system</h1>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(landsepi)</span></code></pre></div>
<div id="initialisation-of-simulation-parameters-seed-time-and-landscape-parameters" class="section level2">
<h2>Initialisation of simulation parameters, seed, time and landscape
parameters</h2>
<p><em>See also the tutorial on <a href="O1-run_simple_simul.html">how
to simulate a simple simulation</a> for details on basic
parameterisation.</em></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Initialisation of the simulation</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">createSimulParams</span>(<span class="at">outputDir =</span> <span class="fu">getwd</span>())</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Seed</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>simul_params<span class="sc">@</span>Seed</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setSeed</span>(simul_params, <span class="at">seed =</span> <span class="dv">1</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>simul_params<span class="sc">@</span>Seed</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Time parameters</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>Nyears <span class="ot">=</span> <span class="dv">6</span> </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>nTSpY <span class="ot">=</span> <span class="dv">120</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setTime</span>(simul_params, <span class="at">Nyears =</span> Nyears, <span class="at">nTSpY =</span> nTSpY)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Inoculum</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setInoculum</span>(simul_params, <span class="at">val =</span> <span class="fl">5e-4</span>)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Landscape</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>landscape <span class="ot">&lt;-</span> <span class="fu">loadLandscape</span>(<span class="at">id =</span> <span class="dv">1</span>)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setLandscape</span>(simul_params, <span class="at">land =</span> landscape)</span></code></pre></div>
<p><em>See also the tutorial on how to <a href="O3-landscape_dispersal.html">parameterise landscape and
dispersal</a> to use your own landscape and compute your own dispersal
matrices.</em></p>
</div>
<div id="setting-the-probability-of-sexual-reproduction" class="section level2">
<h2>Setting the probability of sexual reproduction</h2>
<p>A built-in parameterisation different pathogens (e.g. rusts of cereal
crops) is available using the function <code>loadPathogen()</code>:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>basic_patho_param <span class="ot">&lt;-</span> <span class="fu">loadPathogen</span>(<span class="at">disease =</span> <span class="st">&quot;rust&quot;</span>)</span></code></pre></div>
<p>The built-in parameterisation for “rust” simulates a pathogen which
reproduces purely clonally (repro_sex_prob = 0). It is also possible to
simulate a pathogen which reproduces purely sexually, or both sexually
and clonally at every timestep:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>basic_patho_param<span class="sc">$</span>repro_sex_prob <span class="ot">&lt;-</span> <span class="dv">1</span>  <span class="do">## at every time step all pathogen individuals reproduces sexually</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>basic_patho_param<span class="sc">$</span>repro_sex_prob <span class="ot">&lt;-</span> <span class="dv">0</span>  <span class="do">## at every time step all pathogen individuals reproduces clonally</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>basic_patho_param<span class="sc">$</span>repro_sex_prob <span class="ot">&lt;-</span> <span class="fl">0.5</span> <span class="do">##  at every time step half of the pathogen population</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="do">## reproduce clonally and half sexually</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>basic_patho_param</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $name</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;rust&quot;</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $survival_prob</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1e-04</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $repro_sex_prob</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0.5</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $infection_rate</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0.4</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $propagule_prod_rate</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 3.125</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $latent_period_mean</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 10</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $latent_period_var</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 9</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $infectious_period_mean</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 24</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $infectious_period_var</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 105</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $sigmoid_kappa</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 5.333</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $sigmoid_sigma</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 3</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $sigmoid_plateau</span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1</span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $sex_propagule_viability_limit</span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $sex_propagule_release_mean</span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1</span></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $clonal_propagule_gradual_release</span></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0</span></span></code></pre></div>
<p>To simulate a pathogen with a mixed reproduction system composed of
multiple events of clonal reproduction during the epidemic phase,
followed by a single event of sexual reproduction at the end of the
cropping season, a vector of probabilities of sexual reproduction (one
for each timestep of the cropping season) can be given
(i.e. <code>repro_sex_prob = 0</code> during the epidemic period
<code>[1:nTSpY]</code>, and <code>repro_sex_prob = 1</code> at the end
of the cropping season).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>repro_sex_probs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="fl">0.0</span>, nTSpY), <span class="fl">1.0</span>)  </span></code></pre></div>
<p>The vector containing the probability of sexual reproduction for each
timestep is used to fuel the object <code>simul_params</code> via the
function <code>setReproSexProb()</code>:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setReproSexProb</span>(simul_params, repro_sex_probs)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>simul_params<span class="sc">@</span>ReproSexProb </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [1] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [38] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [75] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [112] 0 0 0 0 0 0 0 0 0 1</span></span></code></pre></div>
</div>
<div id="setting-the-time-of-release-of-propagules-produced-at-the-end-of-a-cropping-season" class="section level2">
<h2>Setting the time of release of propagules produced at the end of a
cropping season</h2>
<div id="sexual-propagules" class="section level3">
<h3>Sexual propagules</h3>
<p>Sexual propagules may be released gradually during the following
seasons (i.e. years). The following parameters set the average number of
cropping seasons after which a sexual propagule is released and the
maximum number of cropping seasons up to which a sexual propagule is
viable:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>basic_patho_param<span class="sc">$</span>sex_propagule_release_mean <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>basic_patho_param<span class="sc">$</span>sex_propagule_viability_limit <span class="ot">=</span> <span class="dv">5</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setPathogen</span>(simul_params, basic_patho_param)  </span></code></pre></div>
<p>Within a cropping season, the day of release of every sexual
propagule is sampled from a uniform distribution with parameters
{0;nTSpY}.</p>
</div>
<div id="clonal-propagules" class="section level3">
<h3>Clonal propagules</h3>
<p>Clonal propagules produced at the end of a cropping season are
released during the following season only, either altogether at the
first day of the season (by setting the parameter
<code>clonal_propagule_gradual_release = FALSE</code>), or progressively
(<code>clonal_propagule_gradual_release = TRUE</code>). In the second
case the day of release of each propagule is sampled from a uniform
distribution with parameters {0;nTSpY}.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>basic_patho_param<span class="sc">$</span>clonal_propagule_gradual_release <span class="ot">=</span> <span class="cn">TRUE</span> <span class="do">## clonal propagules are progressively </span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                                                          <span class="do">## released during the next cropping season</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>basic_patho_param<span class="sc">$</span>clonal_propagule_gradual_release <span class="ot">=</span> <span class="cn">FALSE</span> <span class="do">## clonal propagules are released at </span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>                                                          <span class="do">## the first day of the next cropping season</span></span></code></pre></div>
</div>
</div>
<div id="setting-pathogen-dispersal-both-for-sexual-and-clonal-propagules" class="section level2">
<h2>Setting pathogen dispersal (both for sexual and clonal
propagules)</h2>
<p>Dispersal for both sexual and clonal propagules is given by
vectorised matrices giving the probability of dispersal from any field
of the landscape to any other field. The size of these matrices must be
the square of the number of fields in the landscape. It is thus specific
to both the pathogen and the landscape. For rusts pathogens, built-in
dispersal matrices (for sexual and clonal propagules) are available for
each landscape using the function
<code>loadDispersalPathogen()</code>:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>disp_patho <span class="ot">&lt;-</span> <span class="fu">loadDispersalPathogen</span>(<span class="at">id =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>The first element of the list <code>disp_patho</code> contains the
vectorized dispersal matrix for clonal propagules, while the second
element contains the vectorized dispersal matrix for sexual propagules
(by default this is a diagonal matrix, i.e. sexual propagules spread
only locally and there is no between-field dispersal of sexual
propagules).</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>disp_patho_clonal <span class="ot">&lt;-</span> disp_patho[[<span class="dv">1</span>]]</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>disp_patho_sex <span class="ot">&lt;-</span> disp_patho[[<span class="dv">2</span>]]</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(disp_patho_clonal)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 8.814254e-01 9.525884e-04 7.079895e-10 1.594379e-10 3.285800e-06</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [6] 3.634297e-11</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(disp_patho_sex)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1 0 0 0 0 0</span></span></code></pre></div>
<p>Dispersal matrices can be modified, for example, to simulate a
pathogen whose sexual propagules have the same dispersal ability as
clonal propagules:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>disp_patho_clonal <span class="ot">&lt;-</span> disp_patho[[<span class="dv">1</span>]]</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>disp_patho_sex <span class="ot">&lt;-</span> disp_patho[[<span class="dv">1</span>]]</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(disp_patho_clonal)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 8.814254e-01 9.525884e-04 7.079895e-10 1.594379e-10 3.285800e-06</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [6] 3.634297e-11</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(disp_patho_sex)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 8.814254e-01 9.525884e-04 7.079895e-10 1.594379e-10 3.285800e-06</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [6] 3.634297e-11</span></span></code></pre></div>
<p>Then, the object <code>simul_params</code> is updated with the
dispersal matrices via the function
<code>setDispersalPathogen()</code>:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setDispersalPathogen</span>(simul_params, disp_patho_clonal, disp_patho_sex)</span></code></pre></div>
</div>
<div id="setting-recombination-between-resistance-genes" class="section level2">
<h2>Setting recombination between resistance genes</h2>
<p>For a given parental pair, the genotype of each propagule is issued
from random loci segregation of parental qualitative resistance genes.
For each quantitative resistance gene, the value of each propagule trait
is issued from a normal distribution around the average of the parental
traits, with standard deviation defined by the parameter
<code>recombination_sd</code>, following the infinitesimal model (Fisher
1919).</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Resistance genes</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>gene1 <span class="ot">&lt;-</span> <span class="fu">loadGene</span>(<span class="at">name =</span> <span class="st">&quot;gene 1&quot;</span>, <span class="at">type =</span> <span class="st">&quot;majorGene&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>gene2 <span class="ot">&lt;-</span> <span class="fu">loadGene</span>(<span class="at">name =</span> <span class="st">&quot;gene 2&quot;</span>, <span class="at">type =</span> <span class="st">&quot;QTL&quot;</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co">#gene2$recombination_sd &lt;- 0.8</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>gene2<span class="sc">$</span>Nlevels_aggressiveness <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rbind</span>(gene1, gene2), <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="setting-croptypes-cultivars-and-resistance-genes" class="section level2">
<h2>Setting croptypes, cultivars and resistance genes</h2>
<p>All the other steps (e.g. setting croptypes, cultivars, resistance
genes …) are not impacted by pathogen reproduction system, they are
fully described in <a href="O1-run_simple_simul.html">running a simple
simulation</a></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cultivars</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>cultivar1 <span class="ot">&lt;-</span> <span class="fu">loadCultivar</span>(<span class="at">name =</span> <span class="st">&quot;Susceptible&quot;</span>, <span class="at">type =</span> <span class="st">&quot;growingHost&quot;</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>cultivar2 <span class="ot">&lt;-</span> <span class="fu">loadCultivar</span>(<span class="at">name =</span> <span class="st">&quot;Resistant1&quot;</span>, <span class="at">type =</span> <span class="st">&quot;growingHost&quot;</span>)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>cultivar3 <span class="ot">&lt;-</span> <span class="fu">loadCultivar</span>(<span class="at">name =</span> <span class="st">&quot;Resistant2&quot;</span>, <span class="at">type =</span> <span class="st">&quot;growingHost&quot;</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>cultivars <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rbind</span>(cultivar1, cultivar2, cultivar3)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>                        , <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Allocating genes to cultivars</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setGenes</span>(simul_params, <span class="at">dfGenes =</span> genes)</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setCultivars</span>(simul_params, <span class="at">dfCultivars =</span> cultivars)</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">allocateCultivarGenes</span>(simul_params</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>                                      , <span class="at">cultivarName =</span> <span class="st">&quot;Resistant1&quot;</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>                                      , <span class="at">listGenesNames =</span> <span class="fu">c</span>(<span class="st">&quot;gene 1&quot;</span>))</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">allocateCultivarGenes</span>(simul_params</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>                                      , <span class="at">cultivarName =</span> <span class="st">&quot;Resistant2&quot;</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>                                      , <span class="at">listGenesNames =</span> <span class="fu">c</span>(<span class="st">&quot;gene 2&quot;</span>))</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Allocating cultivars to croptypes</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>croptypes <span class="ot">&lt;-</span> <span class="fu">loadCroptypes</span>(simul_params, <span class="at">names =</span> <span class="fu">c</span>(<span class="st">&quot;Susceptible crop&quot;</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>                                                   , <span class="st">&quot;Resistant crop 1&quot;</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>                                                   , <span class="st">&quot;Resistant crop 2&quot;</span>))</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>croptypes <span class="ot">&lt;-</span> <span class="fu">allocateCroptypeCultivars</span>(croptypes</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>                                       , <span class="at">croptypeName =</span> <span class="st">&quot;Susceptible crop&quot;</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>                                       , <span class="at">cultivarsInCroptype =</span> <span class="st">&quot;Susceptible&quot;</span>)</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>croptypes <span class="ot">&lt;-</span> <span class="fu">allocateCroptypeCultivars</span>(croptypes</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>                                       , <span class="at">croptypeName =</span> <span class="st">&quot;Resistant crop 1&quot;</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>                                       , <span class="at">cultivarsInCroptype =</span> <span class="st">&quot;Resistant1&quot;</span>)</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>croptypes <span class="ot">&lt;-</span> <span class="fu">allocateCroptypeCultivars</span>(croptypes</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>                                       , <span class="at">croptypeName =</span> <span class="st">&quot;Resistant crop 2&quot;</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>                                       , <span class="at">cultivarsInCroptype =</span> <span class="st">&quot;Resistant2&quot;</span>)</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setCroptypes</span>(simul_params, <span class="at">dfCroptypes =</span> croptypes)</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Allocating croptypes to fields of the landscape</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>rotation_sequence <span class="ot">&lt;-</span> croptypes<span class="sc">$</span>croptypeID <span class="do">## No rotation -&gt; 1 rotation_sequence element</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>rotation_period <span class="ot">&lt;-</span> <span class="dv">0</span>  <span class="co"># number of years before rotation of the landscape</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>prop <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>,<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>,<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>)  <span class="co"># proportion (in surface) of each croptype</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>aggreg <span class="ot">&lt;-</span> <span class="dv">0</span>    <span class="co"># level of spatial aggregation</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">allocateLandscapeCroptypes</span>(simul_params</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>                                           , <span class="at">rotation_period =</span> rotation_period</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>                                           , <span class="at">rotation_sequence =</span> rotation_sequence</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>                                           , <span class="at">prop =</span> prop</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>                                           , <span class="at">aggreg =</span> aggreg</span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>                                           , <span class="at">graphic =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Choosing output variables</span></span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>outputlist <span class="ot">&lt;-</span> <span class="fu">loadOutputs</span>(<span class="at">epid_outputs =</span> <span class="st">&quot;all&quot;</span>, <span class="at">evol_outputs =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>simul_params <span class="ot">&lt;-</span> <span class="fu">setOutputs</span>(simul_params, outputlist)</span></code></pre></div>
</div>
<div id="running-the-simulation" class="section level2">
<h2>Running the simulation</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">checkSimulParams</span>(simul_params)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">runSimul</span>(simul_params, <span class="at">graphic =</span> <span class="cn">TRUE</span>, <span class="at">videoMP4 =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
